sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -104,6 +104,56 @@ def on_request(self, request):
         if additional_headers:
             request.http_request.headers.update(additional_headers)
 
+class RequestIdPolicy(SansIOHTTPPolicy):
+    """A simple policy that sets the given request id in the header.
+
+    This will overwrite request id that is already defined in the request. Request id can be
+    configured up front, where the request id will be applied to all outgoing
+    operations, and additional request id can also be set dynamically per operation.
+
+    :param dict base_headers: Headers to send with the request.
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/test_example_sansio.py
+            :start-after: [START request_id_policy]
+            :end-before: [END request_id_policy]
+            :language: python
+            :dedent: 4
+            :caption: Configuring a request id policy.
+    """
+    def __init__(self, request_id=None, **kwargs):  # pylint: disable=super-init-not-called
```

Comment:
Specifying the `request_id` on policy creation is probably not a scenario (I can't come up with a reason to do so). However, specifying the *name* of the header to use may be a scenario. But since adding a new keyword-only constructor parameter is not a breaking change, I'd wait until we actually have a service that needs to do this before introducing it.